File: 3074.apple-redistribution-into-boxes/solution.go
func minimumBoxes(apple []int, capacity []int) int {
    totalApples := 0
    for _, a := range apple {
        totalApples += a
    }
    slices.Sort(capacity)
    i := len(capacity)
    for totalApples > 0 {
        i--
        totalApples -= capacity[i]
    }
    return len(capacity) - i
}

File: 3074.apple-redistribution-into-boxes/solution.kt
class Solution {
    fun minimumBoxes(apple: IntArray, capacity: IntArray): Int {
        var apples = apple.sum()
        capacity.sortDescending()
        val taken = capacity.takeWhile { box -> (apples > 0).also { apples -= box } }
        return taken.size
    }
}